### DISCUSSION TITLE
In memory server

### DISCUSSION THREAD

**Discussion by masterit-technologies** [QuorumDMS/ftp-srv#241] [Q&A] [answered]
Hi everyone, Is it possible to use ftp-srv just for stream redirecting?

I need to catch event STOR and save data into my database, not write on hdd.

Maybe it is stupide question but I dont have any experiences with it.

If its possible can you provide me some short example how to do that?
Thanks.

**Comment by trs** [comment, marked as answer]
To handle file system operations differently, you'd have to implement a [custom file system](https://github.com/autovance/ftp-srv#file-system).
By using the example code as a starting point, you'd have to implement a custom [`write` function](https://github.com/autovance/ftp-srv/blob/master/src/fs.js#L85) that handles your specific logic.

**Reply by masterit-technologies** [reply]
Hi, thanks for the reply, I solved it as you say. I overwrite the write function and used a memory-stream package, then I was able to use the data in memory for my own specific logic.

Something like that.


```
write(fileName, { append, start }) {
        const { fsPath, clientPath } = this._resolvePath(fileName);
        let stream = new MemoryStream(); //memory-stream npm package
        stream.on('finish', function () {
              let data = Buffer.concat(this.buffer).toString('base64'); // data in base64
             //do your work here
        });
        return {
            stream: stream,
            clientPath: clientPath
        };
}
```
